reward/src/lib.rs

```diff
@@ -0,0 +1,524 @@
+        .ok_or_else(|| anyhow!("Couldn't determine distribution strategy"))?;
+
+    let rewards = calculate_rewards(strategy, amount, &all_ens)?;
+
+    // show summary and aggregate payments
+    term::blank();
+    let mut table = term::Table::default();
+    table.push([
+        term::format::bold("Address"),
+        term::format::bold("Reward"),
+        term::format::bold("ENS"),
+    ]);
+    let mut receivers = Vec::new();
+    let mut amounts = Vec::new();
+    for (ens, reward) in rewards.iter() {
+        let resolver: Result<PublicResolver<SignerMiddleware<_, _>>, _> =
+            rt.block_on(PublicResolver::get(ens, signer.clone()));
+
+        // if resolver doesn't exist, we just skip this one
+        if resolver.is_err() {
+            table.push([
+                term::format::italic(term::format::negative("Missing")),
+                term::format::tertiary(u256_to_amount(*reward, decimals)?),
+                term::format::secondary(ens),
+            ]);
+            continue;
+        }
+        let resolver = resolver.unwrap();
+
+        let address = rt
+            .block_on(resolver.address(ens))?
+            .ok_or_else(|| anyhow!("Couldn't get Address of ENS"))?;
+
+        receivers.push(address);
+        amounts.push(*reward);
+
+        table.push([
+            term::format::highlight(address),
+            term::format::tertiary(u256_to_amount(*reward, decimals)?),
+            term::format::secondary(ens),
+        ]);
+    }
+    table.render();
+    term::blank();
+
+    // exit now if this was a dry run
+    if options.dry {
+        return Ok(());
+    }
+
+    if term::confirm("Do you wish to proceed?") {
+        let sum = amounts
+            .iter()
+            .fold(Some(U256::from(0_u64)), |sum, amt| {
+                sum.and_then(|s| s.checked_add(*amt))
+            })
+            .ok_or_else(|| anyhow!("Failed to sum all transactions amounts"))?;
+
+        let token: ERC20<SignerMiddleware<_, _>> = ERC20::new(signer.clone(), address);
+        let spinner = term::spinner(&term::format::tertiary("Checking allowance..."));
+        let allowance = rt.block_on(token.get_allowance(signer.address(), *SUPERSEEDER_ADDRESS))?;
+        spinner.finish();
+
+        if allowance < sum {
+            term::blank();
+            term::info!(
+                "{}",
+                term::format::tertiary("Approving transaction amount for Superseeder...")
+            );
+            term::blank();
+
+            let call = token.approve(*SUPERSEEDER_ADDRESS, sum)?;
+            rt.block_on(term::ethereum::transaction(call))?;
+        }
+
+        // generate tx(s)
+        let superseeder: Superseeder<SignerMiddleware<_, _>> = Superseeder::new(signer);
+        let call = superseeder.send(address, receivers, amounts)?;
+        term::blank();
+        term::info!("{}", term::format::tertiary("Sending transaction..."));
+        term::blank();
+        rt.block_on(term::ethereum::transaction(call))?;
```

Comment:
Poor use of vertical space throughout this function makes it hard to scan for the code you're looking for. You can use vertical space to separate different logical code sections or to group relevant things together:

```rust
        let superseeder: Superseeder<SignerMiddleware<_, _>> = Superseeder::new(signer);
        let call = superseeder.send(address, receivers, amounts)?;

        term::blank();
        term::info!("{}", term::format::tertiary("Sending transaction..."));
        term::blank();

        rt.block_on(term::ethereum::transaction(call))?;
```

This makes it easy to parse the code that actually does stuff vs. the output.

Reply:
I actually use vertical spaces liberally but have been consciously trying to add less of them since most code I come across has less of it than mine.

But since it's okay, I'll be using them more to make the code more readable.

Reply:
Great! Yes in most codebases I come accross the style is "don't waste vertical space" ie. don't use vertical whitespace at all inside functions. I find this really hard to read/skim as someone who reads a lot of code. On the other hand if things form a logical group, they shouldn't have space between them, eg. `let` bindings are best grouped. The pattern I generally use for short to mid-sized functions is:

```
let
let
let

do_stuff();
do_stuff();
do_stuff();

return ...;
```

This breaks things up nicely in a beginning middle and end.